### DISCUSSION TITLE
Compute Power needed

### DISCUSSION THREAD

**Discussion by ywsyws** [Kanaries/graphic-walker#43] [General]
Hi, thank you for creating this. It seems a very interesting tool. May I ask what is the compute power need to run this? I tried to use it but I keep having errors such as "Page unresponsive"  or "File Save Error for [name_of_the_nb]: Invalide response: 413 Request Entity Too Large". Thanks a lot!

**Comment by islxyqwe** [comment]
you can use a duckdb wasm as a compute engine with our dsl parser, [@kanaries/gw-dsl-parser](https://www.npmjs.com/package/@kanaries/gw-dsl-parser)
here is the way that use duckdb in browser:
https://duckdb.org/docs/api/wasm/instantiation
after you instantiated the duckdb, you can use it with below code:
```
import initWasm, { parser_dsl_with_table } from '@kanaries/gw-dsl-parser';
import dslWasm from '@kanaries/gw-dsl-parser/gw_dsl_parser_bg.wasm?url';
/** */
const db = new duckdb.AsyncDuckDB(logger, worker);
const conn = await db.connect();
await db.registerFileText(fileName, JSON.stringify(dataSource));
await conn.insertJSONFromPath(fileName, { name: tableName });
const computation = async (query: any) => {
            const sql = parser_dsl_with_table(tableName, JSON.stringify(query));
            if (process.env.NODE_ENV !== 'production') {
                console.log(query, sql);
            }
            const res = await conn.query(sql).then(transformData);
            return res;
}

/** other codes */
<GraphicWalker computation={computation} rawFields={rawFields} />
```

**Reply by ywsyws** [reply]
Thank you @islxyqwe for your suggestion. Is it possible to provide snippet in Python?

**Reply by ObservedObserver** [reply]
If you want to use graphic-walker in python, check the python version of graphic-walker, which also contains a duckdb engine for computation: https://github.com/Kanaries/pygwalker
